datafusion/src/logical_plan/builder.rs

```diff
@@ -147,10 +147,10 @@ impl LogicalPlanBuilder {
     pub fn scan_parquet_with_name(
         path: impl Into<String>,
         projection: Option<Vec<usize>>,
-        max_concurrency: usize,
+        max_partitions: usize,
         table_name: impl Into<String>,
     ) -> Result<Self> {
-        let provider = Arc::new(ParquetTable::try_new(path, max_concurrency)?);
+        let provider = Arc::new(ParquetTable::try_new(path, max_partitions)?);
```

Comment:
One concern: when increasing the partitions, we also increase the number of maximum nr of threads while reading parquet. I think this should be decoupled.

Reply:
I agree it should be decoupled, but I also don't think this PR makes the coupling any worse -- perhaps we could file a follow on ticket?

Reply:
Filed https://github.com/apache/arrow-datafusion/issues/924 to track